conda_smithy/configure_feedstock.py

```diff
@@ -102,10 +106,10 @@ def render_circle(jinja_env, forge_config, forge_dir):
 
 @contextmanager
 def fudge_subdir(subdir):
-    orig = conda_build.metadata.cc.subdir
-    conda_build.metadata.cc.subdir = subdir
+    orig = cc.subdir
+    cc.subdir = subdir
     yield
-    conda_build.metadata.cc.subdir = orig
+    cc.subdir = orig
 
 
```

Comment:
> I see, so deleting run_docker_build.sh needs to stay.

Right in the case of `pyelftools` the file `run_docker_build.sh` should still be there. It probably just works on Linux, but that's a case where masquerading as a different OS is not required.

> I'll write a test and go from there.

SGTM

Can point out similar cases for all platforms. Might try to write some tests for them over the weekend.

> My opinion on why this is going wrong is that cc is a global variable.  Somewhere in that conda_interface, we're breaking the link to conda, so that there are actually 2 instances floating around.  You are current changing the one in conda-build, but that does not seem to affect the one in conda.

That sounds totally plausible. Really we shouldn't be using `cc` like this if we can avoid it. Though I just don't know what other options we have.